static/app/components/dropdownMenuControlV2.tsx

```diff
@@ -127,8 +127,10 @@ function MenuControl({
   const [triggerWidth, setTriggerWidth] = useState<number>();
   // Update triggerWidth when its size changes using useResizeObserver
   const updateTriggerWidth = useCallback(() => {
-    const newTriggerWidth = ref.current?.offsetWidth;
-    !isSubmenu && newTriggerWidth && setTriggerWidth(newTriggerWidth);
+    setTimeout(() => {
+      const newTriggerWidth = ref.current?.offsetWidth;
+      !isSubmenu && newTriggerWidth && setTriggerWidth(newTriggerWidth);
+    });
```

Comment:
Let's definitely add a comment.

Also I might write this as 


```tsx
  const updateTriggerWidth = useCallback(async() => {
		// XXX: Comment why we do this
    await new Promise(resolve => setTimeout(resolve));

    const newTriggerWidth = ref.current?.offsetWidth;
    !isSubmenu && newTriggerWidth && setTriggerWidth(newTriggerWidth);
  })
```

Reply:
hmm what's the benefit of using a promise over `setTimeout` or @dashed's suggestion `requestAnimationFrame`?

Reply:
Less indentation